sdk/core/Azure.Core/samples/Events.md

```diff
@@ -0,0 +1,190 @@
+# Azure.Core Event samples
+
+**NOTE:** Samples in this file only apply to packages following the
+[Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html).
+The names of these packages usually start with `Azure`.
+
+Most Azure client libraries offer both synchronous and asynchronous methods for
+calling Azure services.  You can distinguish the asynchronous methods by their
+`Async` suffix.  For example, `BlobClient.Download` and `BlobClient.DownloadAsync`
+make the same underlying REST call and only differ in whether they block.  We
+recommend using our async methods for new applications, but there are perfectly
+valid cases for using sync methods as well.  These dual method invocation
+semantics work great, but require a little extra care when writing event handlers.
+
+The `SyncAsyncEventHandler` is a delegate used by events in Azure client
+libraries to represent an event handler that can be invoked from either sync or
+async code paths.  It takes event arguments deriving from `SyncAsyncEventArgs`
+that contain important information for writing your event handler.
+
+- `SyncAsyncEventArgs.CancellationToken` is a cancellation token related to the
+  original operation that raised the event.  It's important for your handler to
+  pass this token along to any asynchronous or long-running synchronous
+  operations that take a token so cancellation (via something like
+  `new CancellationTokenSource(TimeSpan.FromSeconds(10)).Token`, for example)
+  will correctly propagate.
+
+- There is a `SyncAsyncEventArgs.RunSynchronously` flag indicating whether your
+  handler was invoked synchronously or asynchronously.  In general,
+
+    - If you're calling sync methods on your client, you should use sync methods
```

Comment:
It's not clear to me if the reader you're addressing here is the library user or library developer.  Would it be possible to clarify?

Reply:
The whole document is for users of the library.